### PR TITLE
[17.0][FW] base_tier_validation: multiple ports from 14.0

### PR DESCRIPTION
**Pull request by henrybackman**
Port of the following PRs from 14.0 to 17.0:
- https://github.com/OCA/server-ux/pull/421
- https://github.com/OCA/server-ux/pull/412
- ~~https://github.com/OCA/server-ux/pull/426~~ 
- https://github.com/OCA/server-ux/pull/451
- https://github.com/OCA/server-ux/pull/583
- ~~https://github.com/OCA/server-ux/pull/630~~ 
  - ported already in https://github.com/OCA/server-ux/pull/637
- https://github.com/OCA/server-ux/pull/682
- https://github.com/OCA/server-ux/pull/873
- https://github.com/OCA/server-ux/pull/960

### CODE REVIEW

**Review by simahawk — APPROVED** [review]
LGTM, code review only.

**Comment by StefanRijnhart** [comment]
#426 and #630 were already ported to 16 in #637 and are already present in 17 as such.

File: base_tier_validation/models/tier_definition.py [hunk 1]
```diff
@@ -134,7 +134,7 @@ def _get_review_needing_reminder(self):
         return self.env["tier.review"].search(
             [
                 ("definition_id", "=", self.id),
-                ("status", "=", "pending"),
+                ("status", "=", "waiting"),
```
Comment (TDu) [inline base_tier_validation/models/tier_definition.py]: should not this be 
```
("status", "in", ("waiting", "pending",))
```
The waiting status was added and it is not replacing the pending status

**Review by TDu — APPROVED** [review]

**Comment by StefanRijnhart** [comment]
@henrybackman thanks for the thumbs up on my comment, but are you going to remove the commits from your branch?

**Review by LoisRForgeFlow — CHANGES_REQUESTED** [review]
Just blocking the merge while https://github.com/OCA/server-ux/pull/968#issuecomment-2468080707 conversation is resolved

**Comment by henrybackman** [comment]
> @henrybackman thanks for the thumbs up on my comment, but are you going to remove the commits from your branch?

@StefanRijnhart Thank you for reminding me, honestly thought I had already removed them, my bad. They have been removed now.
FYI @LoisRForgeFlow

**Comment by LoisRForgeFlow** [comment]
@StefanRijnhart good from your side now?

**Review by StefanRijnhart — APPROVED** [review]
Thanks!

**Review by LoisRForgeFlow — APPROVED** [review]
/ocabot merge minor

**Pull request merged**